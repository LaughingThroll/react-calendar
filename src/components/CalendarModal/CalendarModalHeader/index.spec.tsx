import { shallow } from 'enzyme'
import { CalendarModalHeader, CalendarModalHeaderProps, ErrorsText } from '.'

const setUp = (props: CalendarModalHeaderProps) => shallow(<CalendarModalHeader {...props} />)

describe('CalendarModalHeader render', () => {
  it('default render', () => {
    const props = { isValidDays: true, countDays: 7 }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes('7')).toBe(true)
  })

  it('invalid render', () => {
    const props = { isValidDays: false, countDays: 7 }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes(ErrorsText.IS_NOT_VALID_DAYS)).toBe(true)
  })
})
