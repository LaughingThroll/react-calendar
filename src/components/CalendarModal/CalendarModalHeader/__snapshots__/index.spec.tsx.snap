// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarModalHeader render default render 1`] = `
<Memo(ModalHeader)
  className={
    Array [
      "calendarModalHeader",
    ]
  }
>
  <h4
    className="calendarModalHeader__title"
  >
    Vacation Request
  </h4>
  <div
    className="calendarModalHeader__days"
  >
     
    7
     Days
  </div>
</Memo(ModalHeader)>
`;

exports[`CalendarModalHeader render invalid render 1`] = `
<Memo(ModalHeader)
  className={
    Array [
      "calendarModalHeader",
    ]
  }
>
  <h4
    className="calendarModalHeader__title"
  >
    Vacation Request
  </h4>
  <div
    className="calendarModalHeader__days"
  >
     
    Is Not Valid
     Days
  </div>
</Memo(ModalHeader)>
`;
